src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -219,6 +219,42 @@ public static IGraphQLBuilder AddSchema<TSchema>(this IGraphQLBuilder builder, F
         }
         #endregion
 
+        #region - AddAutoSchema / WithMutation / WithSubscription -
+        public static IGraphQLBuilderOrConfigureAutoSchema AddAutoSchema<TQueryClrType>(this IGraphQLBuilder builder)
+        {
+            builder.AddSchema<Schema>(ServiceLifetime.Singleton);
+            builder.Services.TryRegister<IGraphTypeMapping, AutoRegisteringGraphTypeMapping>(ServiceLifetime.Singleton);
+            builder.ConfigureSchema((schema, provider) => schema.Query = provider.GetRequiredService<AutoRegisteringObjectGraphType<TQueryClrType>>());
+            return new GraphQLBuilderOrConfigureAutoSchema(builder);
+        }
+
+        public static IGraphQLBuilderOrConfigureAutoSchema WithMutation<TMutationClrType>(this IGraphQLBuilderOrConfigureAutoSchema builder)
```

Comment:
> services.AddGraphQL()
    .AddAutoSchema<Query>()
    .WithMutation<Mutation>()
    .AddSystemTextJson();

Typo?

```
services.AddGraphQL(b =>
   b.AddAutoSchema<Query>()
    .WithMutation<Mutation>()
    .AddSystemTextJson())
```

Also consider to change API as well as we did for `AddGraphQL`:

```
services.AddGraphQL(b =>
   b.AddAutoSchema<Query>(b2 => b2.WithMutation<Mutation>())
    .AddSystemTextJson())
```

The same reasons here as before - more focused API.

Reply:
Yes type-o.

We can, but I was thinking this was simpler and felt more like just registering additional types.  We could also make `b2` be the `Schema`, but I wanted this implementation to be extremely simple, so IntelliSense could easily help people register the mutation and/or subscription CLR types, as that is all that is likely needed for this use case.  Users can always call `ConfigureSchema` for additional configuration on the `Schema` instance.

Finally, we could have `AddAutoSchema` include the mutation or subscription type, but which one?  How to tell which one was intended?  Seemed better to only include the type that is required (TQuery) and use additional methods to configure the mutation and subscription types.

Considering the above, do you still think I should change it like this?

```cs
services.AddGraphQL(b => b
    .AddAutoSchema<Query>(b2 => b2.WithMutation<Mutation>())
    .AddSystemTextJson())
```

Reply:
If you expect that `WithMutation/WithSubscription` methods should always be used with `AddAutoSchema` and "in the context" of `AddAutoSchema` then yes, I would switch to inner configuration delegate. I prefer to group extension methods into internal lambdas, and not to have a flat list. When evolving the API, it helps.

Reply:
Ok.

Reply:
I'll wait until you finish your preliminary review before I make further changes.

Reply:
OK. Today I will not be able to think thoughtfully. I will answer all incoming emails (~10 left) and go to sleep.

Reply:
I looked through code one more time. OK, let's try to go this way. Now, when calculating the graph type from clr type  you have to iterate the entire collection of mappings. Well... OK, it is one time operation when initializing a schema.